data-raw/temperature-data/temp_regression_vignette.Rmd

```diff
@@ -0,0 +1,797 @@
+                     butte_temp_max = max))
+
+# FOR PREDICTIONS identify gaps to predict data
+feather_gap_mean <- feather_lfc_regression_data_full |> 
+  filter(is.na(feather_temp_mean), !is.na(butte_temp_mean)) |> 
+  select(-c(feather_temp_mean, feather_temp_min, feather_temp_max)) |> 
+  rename(butte_temp = butte_temp_mean)
+feather_gap_min <- feather_lfc_regression_data_full |> 
+  filter(is.na(feather_temp_min), !is.na(butte_temp_min)) |> 
+ select(-c(feather_temp_mean, feather_temp_min, feather_temp_max)) |> 
+  rename(butte_temp = butte_temp_min)
+feather_gap_max <- feather_lfc_regression_data_full |> 
+  filter(is.na(feather_temp_max), !is.na(butte_temp_max)) |> 
+ select(-c(feather_temp_mean, feather_temp_min, feather_temp_max)) |> 
+  rename(butte_temp = butte_temp_max)
+
+# FOR MODEL use data where there are no missing data for either butte or feather for regression modeling
+feather_lfc_regression_data_mean <- feather_lfc_regression_data_full |> 
+  filter(!is.na(feather_temp_mean), !is.na(butte_temp_mean)) |> 
+  rename(butte_temp = butte_temp_mean,
+         temp = feather_temp_mean)
+feather_lfc_regression_data_min <- feather_lfc_regression_data_full |> 
+  filter(!is.na(feather_temp_min), !is.na(butte_temp_min)) |> 
+  rename(butte_temp = butte_temp_min,
+         temp = feather_temp_min)
+feather_lfc_regression_data_max <- feather_lfc_regression_data_full |> 
+  filter(!is.na(feather_temp_max), !is.na(butte_temp_max)) |> 
+  rename(butte_temp = butte_temp_max,
+         temp = feather_temp_max)
+```
+
+*Plot of mean temp for Feather River LFC and Butte Creek*
+```{r, echo = F}
+# plot for mean
+ggplot(data = feather_lfc_regression_data_mean, aes(x = butte_temp, y = temp)) +
+  geom_point() +
+  stat_smooth(method = "lm")
+```
+
+*Plot of min temp for Feather River LFC and Butte Creek*
+```{r, echo = F}
+# plot for min
+ggplot(data = feather_lfc_regression_data_min, aes(x = butte_temp, y = temp)) +
+  geom_point() +
+  stat_smooth(method = "lm")
+```
+
+*Plot of max temp for Feather River LFC and Butte Creek*
+```{r, echo = F}
+# plot for max
+ggplot(data = feather_lfc_regression_data_max, aes(x = butte_temp, y = temp)) +
+  geom_point() +
+  stat_smooth(method = "lm")
+```
+
+#### Building Mean Regression
+
+  * Splitting data
+  * Building the Model
+  * Model Summary
+  * Making predictions
+  * Evaluating predictions
+  
+```{r}
+# LFC regression and predictions
+
+# MEAN Regression
+split <-rsample::initial_split(feather_lfc_regression_data_mean, prop = 0.8)
+train <- rsample::training(split)
+test <- rsample::testing(split)
+feather_lfc_mod_mean <- lm(temp ~ date + butte_temp, data = train)
+summary(feather_lfc_mod_mean)
+test_predict <- predict(feather_lfc_mod_mean, test)
+test_predict_df <- test |>
+  mutate(predicted = test_predict)
+# evaluate predictions - MAPE of 10% is not bad
+mean(abs((
+  test_predict_df$predicted - test_predict_df$temp
+)) / test_predict_df$temp)
+```
+
+#### Building Min Regression
+
```

Comment:
Lets work on formatting these sections a little more cleanly. 

The first part of the code sets up and runs the model so you could show that output with some text and then after that chunk include the output of evaluating the model. I don't think we need to show all the code for that and instead can include what the calculated mean absolute error was (might want to look that up to learn what it tells you so can explain it in the text).

This applies for all the min/mean/max sections